ranger/core/main.py

```diff
@@ -436,6 +437,42 @@ def load_custom_commands(*paths):
             ranger.fm = None
 
         allow_access_to_confdir(ranger.args.confdir, False)
+
+        # XXX Load plugins from <ranger.args.datadir>/plugins (experimental)
+        plugindir = os.path.join(ranger.args.datadir, 'plugins')
+        try:
+            os.makedirs(plugindir)
+        except OSError as ex:
+            if ex.errno != errno.EEXIST:
+                raise
+        sys.path.insert(0, plugindir)
+        try:
+            plugins = os.listdir(plugindir)
+        except OSError:
+            LOG.debug('Unable to access plugin directory: %s', plugindir)
+        else:
+            ranger.fm = fm
+            for plugin in sorted(plugins):
```

Comment:
Fine for now, though I'd expect `os.listdir()` to return directory contents in a stable order?

Have you put any thought into topological sorting or something in case people want to be able to write plugins that'll only work if other plugins are already present?

How about filtering this in some way to weed out non-plugins? Seems more foolproof to me that way. Not a big deal though since this is still experimental™.

Reply:
It seems to be in a stable order, just not sorted order.

The topological sorting is an interesting idea. Probably something to think about and come back to if we deem it important. 

Yah I was wondering if I should do any sort of weeding out. Should I filter out any non directories? Might be a good place to start.